lib/nanoc-conref-fs/conref-fs.rb

```diff
@@ -10,22 +13,18 @@ class ConrefFS < Nanoc::DataSources::Filesystem
   # Before iterating over the file objects, this method loads the data folder
   # and applies it to an ivar for later usage.
   def up
-    data_files = NanocConrefFS::Datafiles.collect_data(data_dir_name)
+    data_files = NanocConrefFS::Datafiles.collect_data(ConrefFS.data_dir_name(config))
     NanocConrefFS::Variables.data_files = data_files
     NanocConrefFS::Variables.variables = {}
     reps = @config[:reps] || [:default]
     reps.each { |rep| ConrefFS.load_data_folder(@site_config, rep) }
   end
 
-  def data_dir_name
-    config.fetch(:data_dir) { |_| 'data' }
-  end
-
   def self.load_data_folder(config, rep)
     return unless NanocConrefFS::Variables.variables[rep].nil?
     data_files = NanocConrefFS::Variables.data_files
     data = NanocConrefFS::Datafiles.process(data_files, config, rep)
-    NanocConrefFS::Variables.variables[rep] = { 'site' => { 'config' => config, 'data' => data } }
+    NanocConrefFS::Variables.variables[rep] = { 'site' => { 'config' => config, data_dir_name => data } }
```

Comment:
Why did you need to change this to data_dir_name?

Reply:
Half of the chain is based on the file path, the other is based on the hash path.  The two intersect and over-lap throughout the library (using the file path to lookup values in the hash) and must be made to be identical.  :/